### PR TITLE
feat(blocks): support delete hotkey

### LINKED ISSUE #2550: Please support the `Delete` key to function the same way as the `Backspace` key.
### What happened?

reproduction video as below.

https://github.com/toeverything/AFFiNE/assets/54631354/63d853a3-35c0-4e6c-bd80-2f651e5845a7

### Version

macOS x64

### What browsers are you seeing the problem on?

_No response_

### Relevant log output

_No response_

### Anything else?

AFFINE should support `Delete` key to allow user **delete** their blocks
version: 0.6.0-canary.0

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by Alkaidcc**
fix #2550

### CODE REVIEW

**Comment by lawvs** [comment]
Do you have time to add some tests for it?

**Comment by Alkaidcc** [comment]
I'll think about it. I discovered that `delete` key in BlockSuite does not working as expected(like notion/wolai). I might need more time to make it work properly.😢

**Comment by lawvs** [comment]
> I'll think about it. I discovered that `delete` key in BlockSuite does not working as expected(like notion/wolai). I might need more time to make it work properly.😢

If you have any questions, feel free to ask us at any time.

File: packages/blocks/src/__internal__/utils/common-operations.ts [hunk 1]
```diff
@@ -42,6 +42,11 @@ export function isCollapsedAtBlockStart(vEditor: VEditor) {
   return vRange?.index === 0 && vRange?.length === 0;
 }
 
+export function isCollapsedAtBlockEnd(vEditor: VEditor) {
+  const vRange = vEditor.getVRange();
+  return vRange?.index === vEditor.yText.length;
```
Comment (ahonn) [inline packages/blocks/src/__internal__/utils/common-operations.ts:47]: Missing condition `vRange.length === 0` check to make sure range is collapsed

Comment (Alkaidcc) [reply packages/blocks/src/__internal__/utils/common-operations.ts:47]: Sorry, I don't understand **collapsed** here. This code means checking the current cursor at the end of this line, in this case, ` vRange.length === 0` is always `true`.
If you want to help, PR is welcome!

Comment (ahonn) [reply packages/blocks/src/__internal__/utils/common-operations.ts:47]: collapsed means that the range is a cursor, otherwise it is not.

**Comment by Alkaidcc** [comment]
Could you review my code in your spare time? 👀  @lawvs

**Pull request merged**